types/react-map-gl/index.d.ts

```diff
@@ -423,3 +424,47 @@ export interface SVGOverlayProps extends BaseControlProps {
 }
 
 export class SVGOverlay extends BaseControl<SVGOverlayProps, Element> {}
+
+export interface SourceProps {
```

Comment:
Where did you get all these props from? In the documentation and the source code, I only see `id`, `type`, and `children`. Or am I missing something?

Reply:
In the source code of react-map-gl ,sourceOptions is provdied implictly and extra properties will be added through the loop.
https://github.com/uber/react-map-gl/blob/c71b0d4b9da65bdeaa4a077d85b4658654b68100/src/components/source.js#L96
I add these extra properties and set them as optional and people could pass extra properties which are defined by mapbox-gl with type checking.

Reply:
Ok, cool